polyfill/lib/datetime.mjs

```diff
@@ -361,8 +361,13 @@ export class DateTime {
   difference(other, options) {
     if (!ES.IsTemporalDateTime(this)) throw new TypeError('invalid receiver');
     if (!ES.IsTemporalDateTime(other)) throw new TypeError('invalid DateTime object');
-    const calendar = GetSlot(this, CALENDAR);
-    if (calendar.id !== GetSlot(other, CALENDAR).id) {
+    const thisCalendar = GetSlot(this, CALENDAR);
+    const otherCalendar = GetSlot(other, CALENDAR);
+    const calendar = ES.ChooseCommonCalendar(thisCalendar, otherCalendar);
+    if (!calendar) {
+      throw new RangeError(`cannot compute difference between dates of ${thisCalendar} and ${otherCalendar} calendars`);
+    }
+    if (calendar.id !== otherCalendar.id) {
```

Comment:
Do you also need `if (calendar.id !== thisCalendar.id) {` like you wrote in date.mjs?

Reply:
We don't; this is because we create a new Temporal.Date from the date portion of the DateTime even if the calendars do match, in order to pass to `calendar.dateDifference()`.